### PR TITLE
feat(cli): improve lb4 update

### PR DESCRIPTION
**Pull request by raymondfeng**
We check the existence of `@loopback/core` only in `dependencies` as of today. It should be expanded to `devDependencies` and `peerDependencies` as we push some of the loopback deps to such sections in package.json

This PR also allows any `@loopback/*` packages to be updated.

1. Allow `lb4 update` to run against any projects that use `@loopback/*` in dependencies, devDependencies, or peerDependencies.

2. Fix the CLI template.

## Checklist

- [x] DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [x] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: packages/cli/test/integration/lib/check-loopback-project.suite.js [hunk 1]
```diff
@@ -73,6 +73,26 @@ module.exports = function suiteCheckLoopBackProject(generator) {
...
+    testCheckLoopBack(
+      'throws an error if "@loopback/core" is only a peer dependency',
+      {
+        peerDependencies: {
+          '@loopback/core': coreVer,
+        },
+      },
+      /No `@loopback\/core` package found/,
+    );
+
```
Comment (agnes512) [inline packages/cli/test/integration/lib/check-loopback-project.suite.js:95]: What't the differences between these two test cases and those 2 cases below(`'passes if "@loopback/core" is a dev dependency'` and `'passes if "@loopback/core" is a peer dependency'`)?

Comment (raymondfeng) [reply packages/cli/test/integration/lib/check-loopback-project.suite.js:95]: For a package, we can declare regular and peer dependencies. These two tests make sure we allow `@loopback/*` in both cases.

Comment (jannyHou) [reply packages/cli/test/integration/lib/check-loopback-project.suite.js:95]: I think @agnes512 was asking about these two and the ones from line 242. 
There two test with default options (decision: abort), the next two are with options `command: update`

Comment (raymondfeng) [reply packages/cli/test/integration/lib/check-loopback-project.suite.js:95]: The `command: 'update'` indicates the library is used with `lb4 update` command while others are part of any `lb4` command that expects a LB application project.

Comment (jannyHou) [reply packages/cli/test/integration/lib/check-loopback-project.suite.js:95]: I see, thanks for the explanation 👍

**Review by jannyHou — APPROVED** [review]
👍 LGTM.

**Comment by agnes512** [comment]
👍

**Pull request merged**